pkg/koordlet/metricsadvisor/collectors/prom/collect_rule.go

```diff
@@ -0,0 +1,235 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prom
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"gopkg.in/yaml.v2"
+	"k8s.io/klog/v2"
+)
+
+const (
+	DefaultScrapeInterval   = 30 * time.Second
+	DefaultScrapeTimeout    = 10 * time.Second
+	MinValidMetricTimeRange = 2 * time.Second
+)
+
+// CollectRule is the collecting rule of the scrapers.
+// It defines whether the global scraper is enabled and the jobs to be scraped.
+/**
+# Example
+config:
+  enable: true
+  jobs:
+  - name: job1
+    enable: true
+    address: "127.0.0.1"
+    port: 9317
+    path: /metrics
+    scrapeInterval: 10s
+    scrapeTimeout: 5s
+    metrics:
+    - sourceName: metric1
+      sourceLabels:
+        label1: value1
+      targetName: metric1
+    - sourceName: metric2
+      targetName: metricA
+  - name: job2
+    enable: true
+    address: "127.0.0.1"
+    port: 9316
+    path: /metrics
+    scrapeInterval: 30s
+    scrapeTimeout: 10s
+    metrics:
+    - sourceName: metric3
+      targetName: metric3
+  - name: job3
+    enable: false
+    address: "127.0.0.1"
+    port: 9090
+    path: /metrics
+    scrapeInterval: 30s
+    scrapeTimeout: 10s
+    metrics:
+    - sourceName: metric3
+      targetName: metric3
+**/
+type CollectRule struct {
+	Enable bool      `json:"enable,omitempty"`
+	Jobs   []JobRule `json:"jobs,omitempty"`
+}
+
+// JobRule is the rule of the job.
+// It defines the job name, whether the job is enabled, the job address, port, path, scrape interval, scrape timeout,
+// and the metrics to be scraped.
+type JobRule struct {
+	Name           string        `yaml:"name,omitempty" json:"name,omitempty"`
+	Enable         bool          `yaml:"enable,omitempty" json:"enable,omitempty"`
+	Address        string        `yaml:"address,omitempty" json:"address,omitempty"`
+	Port           int           `yaml:"port,omitempty" json:"port,omitempty"`
+	Path           string        `yaml:"path,omitempty" json:"path,omitempty"`
+	ScrapeInterval time.Duration `yaml:"scrapeInterval,omitempty" json:"scrapeInterval,omitempty"`
+	ScrapeTimeout  time.Duration `yaml:"scrapeTimeout,omitempty" json:"scrapeTimeout,omitempty"`
+	Metrics        []MetricRule  `yaml:"metrics,omitempty" json:"metrics,omitempty"`
+}
+
+// MetricRule is the rule of the metric.
+// It defines the name of the source metric to be scraped, the matched labels of the source metric and the name of
+// the target metric to be stored into the metric cache.
+type MetricRule struct {
+	SourceName   string         `yaml:"sourceName,omitempty" json:"sourceName,omitempty"`
+	SourceLabels model.LabelSet `yaml:"sourceLabels,omitempty" json:"sourceLabels,omitempty"`
+	TargetName   string         `yaml:"targetName,omitempty" json:"targetName,omitempty"`
+	TargetType   MetricType     `yaml:"targetType,omitempty" json:"targetType,omitempty"`
```

Comment:
the label to identify id of node/pod/container

Reply:
added